### PR TITLE
AVRO-3335 Evaluate input of GenerateNames

### PR DESCRIPTION
**Pull request by KyleSchoonover**
Additionally updated naming since it does not need to preserve Java like naming.

### Jira

- [ ] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-3335

### Tests

- [ ] My PR adds the following unit tests: Test that method throws new exception

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: lang/csharp/src/apache/main/CodeGen/CodeGen.cs [hunk 1]
```diff
@@ -219,8 +219,14 @@ protected virtual void processProtocols()
         /// <returns>
         /// List of named schemas.
         /// </returns>
-        protected virtual SchemaNames generateNames(Protocol protocol)
+        /// <exception cref="System.ArgumentNullException">protocol - Protocol can not be null.</exception>
+        protected virtual SchemaNames GenerateNames(Protocol protocol)
```
Comment (martin-g) [inline lang/csharp/src/apache/main/CodeGen/CodeGen.cs:223]: For backward compatibility we need to keep the old method deprecated for few releases.

File: lang/csharp/src/apache/main/CodeGen/CodeGen.cs [hunk 2]
```diff
@@ -237,7 +243,7 @@ protected virtual SchemaNames generateNames(Protocol protocol)
         /// <returns>
         /// List of named schemas.
         /// </returns>
-        protected virtual SchemaNames generateNames(Schema schema)
+        protected virtual SchemaNames GenerateNames(Schema schema)
```
Comment (martin-g) [inline lang/csharp/src/apache/main/CodeGen/CodeGen.cs:246]: As above.

File: lang/csharp/src/apache/main/CodeGen/CodeGen.cs [hunk 3]
```diff
@@ -219,8 +219,14 @@ protected virtual void processProtocols()
         /// <returns>
         /// List of named schemas.
         /// </returns>
-        protected virtual SchemaNames generateNames(Protocol protocol)
+        /// <exception cref="System.ArgumentNullException">protocol - Protocol can not be null.</exception>
+        protected virtual SchemaNames GenerateNames(Protocol protocol)
         {
+            if (protocol == null)
+            {
+                throw new ArgumentNullException(nameof(protocol), "Protocol can not be null");
```
Comment (martin-g) [inline lang/csharp/src/apache/main/CodeGen/CodeGen.cs:227]: What does `nameof()` returns for `null` ?

**Review by martin-g — CHANGES_REQUESTED** [review]

File: lang/csharp/src/apache/main/CodeGen/CodeGen.cs (hunk 3 above)
Comment (KyleSchoonover) [reply lang/csharp/src/apache/main/CodeGen/CodeGen.cs:227]: Honestly, it's a weird pattern, but really it's to make sure your messaging is up to date when you change the name of a variable.  It's the equivalent of "protocol."  It's literally the name of the variable as a string.

File: lang/csharp/src/apache/main/CodeGen/CodeGen.cs (hunk 1 above)
Comment (KyleSchoonover) [reply lang/csharp/src/apache/main/CodeGen/CodeGen.cs:237]: Fixed

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @KyleSchoonover !

**Pull request merged**